libs/common/result.hpp

```diff
@@ -187,36 +206,119 @@ namespace iroha {
       return Error<E>{std::forward<E>(error)};
     }
 
+    template <typename T>
+    constexpr bool isResult = std::is_base_of<ResultBase, T>::value;
+    template <typename T>
+    constexpr bool isValue = std::is_base_of<ValueBase, T>::value;
+    template <typename T>
+    constexpr bool isError = std::is_base_of<ErrorBase, T>::value;
+
+    /**
+     * A struct that provides the result type conversion for bind operator.
+     * @tparam Transformed The type returned by value transformation function.
+     * @tparam ErrorType The type of former result's error
+     * The struct provides Result type, which is a combination of transformation
+     * function outcome and former Result error type, and a method to convert
+     * transformation function outcome to this type.
+     */
+    template <typename Transformed, typename ErrorType, typename = void>
```

Comment:
`typename = void` - is it needed?

Reply:
it is needed for partial specialisation (please see below)